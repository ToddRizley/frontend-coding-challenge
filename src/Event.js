import React, { Component } from 'react';
import { formatDate } from './utils.js';

class Event extends Component {

  render() {
    return (
      <div className='row col-xs-12'>
        {this.props.eventInfo.title} | From: {formatDate(this.props.eventInfo.start_time)} To: {formatDate(this.props.eventInfo.end_time)}
      </div>
  )};
};

export default Event;
