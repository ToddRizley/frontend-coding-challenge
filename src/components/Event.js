import React, { Component } from 'react';
import { formatDate } from '../utils.js';

export default class Event extends Component {

  render() {
    return (
      <div className='row col-xs-12'>
        {this.props.eventInfo.title} | From: {formatDate(this.props.eventInfo.start_time)} To: {formatDate(this.props.eventInfo.end_time)}
      </div>
  )};
};

Event.propTypes = {
  eventInfo: React.PropTypes.object
};
