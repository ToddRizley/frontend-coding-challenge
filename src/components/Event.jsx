import React from 'react';
import { formatDate } from '../utils';

const Event = props => (
  <div className="row col-xs-12">
    {props.eventInfo.title}&nbsp;
        |  From: {formatDate(props.eventInfo.start_time)}
        To: {formatDate(props.eventInfo.end_time)}
  </div>
);

Event.propTypes = {
  eventInfo: React.PropTypes.object,
};

export default Event;
