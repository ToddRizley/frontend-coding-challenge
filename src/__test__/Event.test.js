import React from 'react';
import Event from '../Event'
import renderer from 'react-test-renderer';

describe('<Event/>', () => {
   it('renders correctly', () => {
    const eventInfo = { id: 1, title: 'Event One', start_time: '7:00 PM 11-24-16', end_time: '9:00 PM 11-24-16' };
    const rendered = renderer.create(
      <Event key={eventInfo.id} eventInfo={eventInfo} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
