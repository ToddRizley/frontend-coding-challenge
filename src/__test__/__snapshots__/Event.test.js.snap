// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Event/> renders correctly 1`] = `
<div
  className="row col-xs-12"
>
  Event One
   | From: 
  7:00 PM 11-24-16
   To: 
  9:00 PM 11-24-16
</div>
`;
